src/NServiceBus.Core/Routing/INonCompetingConsumersTransport.cs

```diff
@@ -0,0 +1,9 @@
+﻿namespace NServiceBus.Routing
+{
+    /// <summary>
+    /// Marks the transport as unable to use competing consumers pattern.
+    /// </summary>
+    public interface INonCompetingConsumersTransport
```

Comment:
To me it is about intent and logical coupling. I believe it is wrong to refer to MSMQ directly in the routing code that is supposed to be transport-neutral. 

Alternatively we move all the stuff that is exposed to `INonCompetingConsumersTransport` transports to the MSMQ folder to make it obvious it is MSMQ-only.